aiosmtpd/tests/test_smtp.py

```diff
@@ -790,3 +792,87 @@ def test_mail_invalid_body_param(self):
             self.assertEqual(
                 response,
                 b'Error: BODY can only be one of 7BIT, 8BITMIME')
+
+
+class TestClientCrash(unittest.TestCase):
+    # GH#62 - if the client crashes during the SMTP dialog we want to make
+    # sure we don't get tracebacks where we call readline().
+    def setUp(self):
+        controller = Controller(Sink)
+        controller.start()
+        self.addCleanup(controller.stop)
+        self.address = (controller.hostname, controller.port)
+
+    def test_connection_reset_during_DATA(self):
+        with SMTP(*self.address) as client:
+            client.helo('example.com')
+            client.docmd('MAIL FROM: <anne@example.com>')
+            client.docmd('RCPT TO: <bart@example.com>')
+            client.docmd('DATA')
+            # Start sending the DATA but reset the connection before that
+            # completes, i.e. before the .\r\n
+            client.send(b'From: <anne@example.com>')
+            reset_connection(client)
+            # The connection should be disconnected, so trying to do another
+            # command from here will give us an exception.  In GH#62, the
+            # server just hung.
+            self.assertRaises(SMTPServerDisconnected, client.noop)
+
+    def test_connection_reset_during_command(self):
+        with SMTP(*self.address) as client:
+            client.helo('example.com')
+            # Start sending a command but reset the connection before that
+            # completes, i.e. before the \r\n
+            client.send('MAIL FROM: <anne')
```

Comment:
I think this should be a bytestring for consistency with the other tests. (Does `SMTP.send` even accept a str?)